kafka/src/test/java/org/axonframework/extensions/kafka/eventhandling/util/KafkaContainerCluster.java

```diff
@@ -0,0 +1,107 @@
+
+    public KafkaContainerCluster(String confluentPlatformVersion, int brokersNum, int internalTopicsRf) {
+        if (brokersNum < 0) {
+            throw new IllegalArgumentException("brokersNum '" + brokersNum + "' must be greater than 0");
+        }
+        if (internalTopicsRf < 0 || internalTopicsRf > brokersNum) {
+            throw new IllegalArgumentException(
+                    "internalTopicsRf '" + internalTopicsRf + "' must be less than brokersNum and greater than 0");
+        }
+
+        this.brokersNum = brokersNum;
+        this.network = Network.newNetwork();
+
+        this.zookeeper = new GenericContainer<>(DockerImageName.parse("confluentinc/cp-zookeeper")
+                                                               .withTag(confluentPlatformVersion))
+                .withNetwork(network)
+                .withNetworkAliases("zookeeper")
+                .withEnv("ZOOKEEPER_CLIENT_PORT", String.valueOf(KafkaContainer.ZOOKEEPER_PORT));
+
+        this.brokers = IntStream
+                .range(0, this.brokersNum)
+                .mapToObj(brokerNum -> new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka")
+                                                                         .withTag(confluentPlatformVersion))
+                        .withNetwork(this.network)
+                        .withNetworkAliases("broker-" + brokerNum)
+                        .dependsOn(this.zookeeper)
+                        .withExternalZookeeper("zookeeper:" + KafkaContainer.ZOOKEEPER_PORT)
+                        .withEnv("KAFKA_BROKER_ID", brokerNum + "")
+                        .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", internalTopicsRf + "")
+                        .withEnv("KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS", internalTopicsRf + "")
+                        .withEnv("KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", internalTopicsRf + "")
+                        .withEnv("KAFKA_TRANSACTION_STATE_LOG_MIN_ISR", internalTopicsRf + ""))
+                .collect(Collectors.toList());
+    }
+
+    public Collection<KafkaContainer> getBrokers() {
+        return this.brokers;
+    }
+
+    public String getBootstrapServers() {
+        return brokers.stream()
+                      .map(KafkaContainer::getBootstrapServers)
+                      .collect(Collectors.joining(","));
+    }
+
+    private Stream<GenericContainer<?>> allContainers() {
+        return Stream.concat(
+                this.brokers.stream(),
+                Stream.of(this.zookeeper)
+        );
+    }
+
+    @Override
+    public void start() {
+        Stream<Startable> startables = this.brokers.stream().map(Startable.class::cast);
+        try {
+            Startables.deepStart(startables).get(60, SECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            e.printStackTrace();
+        }
+        Unreliables.retryUntilTrue(30, TimeUnit.SECONDS, () -> {
```

Comment:
Curious: is this because ZooKeeper is unreliable during start-up? Or are you guys uncertain as this has been copied from a different location? ;-)